lib/puppet/provider/snmp_notification_receiver/cisco_nexus.rb

```diff
@@ -0,0 +1,85 @@
+require 'puppet/resource_api/simple_provider'
+
+# Implementation for the snmp_notification_receiver type using the Resource API.
+class Puppet::Provider::SnmpNotificationReceiver::CiscoNexus < Puppet::ResourceApi::SimpleProvider
+  def get(_context, receivers=nil)
+    require 'cisco_node_utils'
+    current_states = []
+    if receivers.nil? || receivers.empty?
+      @snmp_notification_receivers ||= Cisco::SnmpNotificationReceiver.receivers
+      @snmp_notification_receivers.each do |receiver, instance|
+        current_states << {
```

Comment:
To much duplicate code here break this out to a function. Which will require a rejig of the unit tests.